src/Microsoft.AspNetCore.Razor.Evolution/TagHelperBinderSyntaxTreePass.cs

```diff
@@ -70,6 +77,228 @@ public RazorSyntaxTree Execute(RazorCodeDocument codeDocument, RazorSyntaxTree s
+            var text = directive.DirectiveText;
+            var lookupStrings = text?.Split(new[] { ',' });
+
+            // Ensure that we have valid lookupStrings to work with. The valid format is "typeName, assemblyName"
+            if (lookupStrings == null ||
+                lookupStrings.Any(string.IsNullOrWhiteSpace) ||
+                lookupStrings.Length != 2)
+            {
+                errorSink.OnError(
+                    directive.Location,
+                    Resources.FormatInvalidTagHelperLookupText(text),
+                    Math.Max(text.Length, 1));
+
+                return null;
+            }
+
+            var trimmedAssemblyName = lookupStrings[1].Trim();
+
+            // + 1 is for the comma separator in the lookup text.
+            var assemblyNameIndex =
+                lookupStrings[0].Length + 1 + lookupStrings[1].IndexOf(trimmedAssemblyName, StringComparison.Ordinal);
+            var assemblyNamePrefix = directive.DirectiveText.Substring(0, assemblyNameIndex);
+            var assemblyNameLocation = new SourceLocation(
+                directive.Location.FilePath,
+                directive.Location.AbsoluteIndex + assemblyNameIndex,
+                directive.Location.LineIndex,
+                directive.Location.CharacterIndex + assemblyNameIndex);
+
+            return new ParsedDirective
+            {
+                TypePattern = lookupStrings[0].Trim(),
+                AssemblyName = trimmedAssemblyName,
+                AssemblyNameLocation = assemblyNameLocation,
+            };
+        }
+
+        // Internal for testing
+        internal bool IsValidTagHelperPrefix(
+            string prefix,
+            SourceLocation directiveLocation,
+            ErrorSink errorSink)
+        {
+            foreach (var character in prefix)
+            {
+                // Prefixes are correlated with tag names, tag names cannot have whitespace.
+                if (char.IsWhiteSpace(character) ||  InvalidNonWhitespaceNameCharacters.Contains(character))
+                {
+                    errorSink.OnError(
+                        directiveLocation,
+                        Resources.FormatInvalidTagHelperPrefixValue(
+                            SyntaxConstants.CSharp.TagHelperPrefixKeyword,
+                            character,
+                            prefix),
+                        prefix.Length);
+
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static IReadOnlyList<TagHelperDescriptor> PrefixDescriptors(
+            string prefix,
+            IEnumerable<TagHelperDescriptor> descriptors)
+        {
+            if (!string.IsNullOrEmpty(prefix))
+            {
+                return descriptors.Select(descriptor => new TagHelperDescriptor(descriptor)
+                {
+                    Prefix = prefix
+                }).ToList();
+            }
+
+            return descriptors.ToList();
```

Comment:
Yeah do reason to, I think this was from an earlier version where `descriptors` had a different type.